packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -11,14 +16,108 @@ const Separator = styled(Toolbar.Separator)`
   height: ${pxToRem(24)};
 `;
 
+/**
+ *
+ * @param {import('slate').BaseEditor} editor
+ * @param {string} name - name of the modifier
+ */
+const isModifierActive = (editor, name) => {
+  const modifiers = Editor.marks(editor);
+
+  return modifiers ? modifiers[name] === true : false;
+};
+
+/**
+ *
+ * @param {import('slate').BaseEditor} editor
+ * @param {string} name - name of the modifier
+ */
+const toggleModifier = (editor, name) => {
+  const isActive = isModifierActive(editor, name);
+
+  if (isActive) {
+    Editor.removeMark(editor, name);
+  } else {
+    Editor.addMark(editor, name, true);
+  }
+};
+
+const ModifierButton = ({ icon, name, label, editor }) => {
+  const isActive = isModifierActive(editor, name);
+  const { formatMessage } = useIntl();
+
+  const labelMessage = formatMessage(label);
+
+  return (
+    <Tooltip description={labelMessage}>
+      <Toolbar.ToggleItem value={name} data-state={isActive ? 'on' : 'off'} asChild>
+        <Flex
+          background={isActive ? 'primary100' : ''}
+          padding={2}
+          as="button"
+          hasRadius
+          onClick={() => {
+            toggleModifier(editor, name);
+          }}
+          aria-label={labelMessage}
+        >
+          <Icon width={4} as={icon} color={isActive ? 'primary600' : 'neutral600'} />
+        </Flex>
+      </Toolbar.ToggleItem>
+    </Tooltip>
+  );
+};
+
+ModifierButton.propTypes = {
+  icon: PropTypes.elementType.isRequired,
+  name: PropTypes.string.isRequired,
+  label: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    defaultMessage: PropTypes.string.isRequired,
+  }).isRequired,
+  editor: PropTypes.object.isRequired,
+};
+
+const modifiers = [
+  {
+    name: 'bold',
+    icon: Bold,
+    label: { id: 'components.Blocks.modifiers.bold', defaultMessage: 'Bold' },
+  },
+  {
+    name: 'italic',
+    icon: Italic,
+    label: { id: 'components.Blocks.modifiers.italic', defaultMessage: 'Italic' },
+  },
+  {
+    name: 'underline',
+    icon: Underline,
+    label: { id: 'components.Blocks.modifiers.underline', defaultMessage: 'Underline' },
+  },
+  {
+    name: 'strikethrough',
+    icon: StrikeThrough,
+    label: { id: 'components.Blocks.modifiers.strikethrough', defaultMessage: 'Strikethrough' },
+  },
+];
+
 const BlocksToolbar = () => {
+  const editor = useSlate();
```

Comment:
what benefits do you see calling this hook here and prop-drilling as opposed to using in the specific component? :)

Reply:
I figured it would avoid calling the hook 4 times, but I guess it's not worth polluting the props. I'll update